pysoa/server/standalone.py

```diff
@@ -19,14 +22,63 @@
     exit(99)
 
 
+def _get_arg_parser():
+    import argparse
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '-f', '--fork-processes', '--fork',
+        help='The number of processes to fork (if 0, 1, or none, no process is forked; the server is run directly)',
+        required=False,
+        type=int,
+        default=0,
+    )
+    return parser
+
+
+def _get_args(parser):
+    return parser.parse_known_args()[0]
+
+
+def _run_server(args, server_class):
+    if args.fork_processes > 1:
+        import multiprocessing
+        import signal
+        import time
+
+        processes = []
+        cmd = {'keep_starting': True}
+
+        def _stop(*_):
+            cmd['keep_starting'] = False
+
+        signal.signal(signal.SIGINT, _stop)
+        signal.signal(signal.SIGTERM, _stop)
```

Comment:
you're only using these signals _only_ to stop startup, and you're not removing the handlers after the startup is done.

Reply:
Yes, this was actually quite intentional. In fact, the examples I found online said that the parent process needs to set the signal to `signal.SIG_IGN` or some other function that ignores it, because that can cause the parent process to stop `.join`-ing and it won’t wait before the child processes stop before stopping itself. I did it this way to be able to stop the startup process.